### PR TITLE
split architecture lib/utils

### PR DESCRIPTION
**Pull request by sjwahub**
Signed-off-by: Shujun Wang <shujun.wang@intel.com>

Initial draft  

unit test log: 
[SplitArcUnitTestCase.txt](https://github.com/ipdk-io/ptf-tests/files/10045405/SplitArcUnitTestCase.txt)

### CODE REVIEW

**Review by stadju — APPROVED** [review]

**Review by dkaranjai — CHANGES_REQUESTED** [review]
General Comment: Please add comments to all test case steps

**Comment by dkaranjai** [comment]
Brief of changes: 

 

Lib changes 

 1.1   New files "p4rt_ctl.py" and "p4rt-ctl.in" are introduced  

 1.2   "port_config.py" is modified by changing all "gnmi_cli" to 'gnmi_ctl' 

 1.3   Feature specific changes  

 

Utils changes 

2.1   "gnmi_ctl_utils.py" is to replace "gnmi_cli.utils.py" 

2.2    "p4rtctl_utils.py" is to replace "ovsp4ctl_utils.py" 

2.3    new utils "log.py" is introduced with timestamp  

2.4    Delete old "p4ovs_utils.py" and "gnmi_cli_utils.py" 

2.5    Feature specific changes  

 

Test Cases changes 

3.1  import statement need changes 

3.2  All p4ovs-ctl and gnmi_cli need to be replaced  in all test cases with p4rt_ctl and gnmi_ctl respectively  

3.3  All  print() for logging purpose, except for usage purpose, should be replaced by log.passed(), log.info(), log.failed() and log.warning() accordingly 

3.4  There are total 66 test cases in https://github.com/ipdk-io/ptf-tests.  [ Additional 6 ECMP FRR case need to be moved to this location ] 

In which, 31 test cases need to call new utils "test_utils.gen_dep_files_p4c_tdi_pipeline_builder()" instead of old "test_utils.gen_dep_files_p4c_ovs_pipeline_builder()" 

For remaining test cases, old function name "gen_dep_files_p4c_dpdk_pna_ovs_pipeline_builder()" with new code is still being used. We need to rename it to  "gen_dep_files_p4c_dpdk_pna_tdi_pipeline_builder()" 

 

3.5  All test cases are impacted by following utils functions (previously p4ovs_ctl, now p4rt_ctl) 

 add-entry SWITCH TABLE FLOW           adds new table entry 

 del-entry SWITCH TABLE KEY                 deletes table entry with KEY from TABLE 

 dump-entries SWITCH [TBL]                   prints table entries 

 set-default-entry SWITCH TBL ACTION       sets default table entry for TBL 

 get-default-entry SWITCH TBL          prints default table entry for TBL 

 add-action-profile-member SWITCH PROFILE FLOW      adds member reference 

 delete-action-profile-member SWITCH PROFILE FLOW      deletes member reference 

 get-action-profile-member SWITCH PROFILE FLOW      prints member entries 

 add-action-profile-group SWITCH PROFILE FLOW      adds group reference 

 delete-action-profile-group SWITCH PROFILE FLOW      deletes group reference 

 get-action-profile-group SWITCH PROFILE FLOW      prints group entries 

 get-counter SWITCH CTR_TBL CTR_FLOW       gets indirect counter table entry 

 reset-counter SWITCH CTR_TBL CTR_FLOW     resets counter table entry 

 

3.6 All test case formatting related changes 

Documentation 

Python Typing  

Python Lint -- Black

**Review by stadju — APPROVED** [review]

File: tests/DPDK_Hot_Plug_multi_port_test.py [hunk 1]
```diff
@@ -63,54 +64,62 @@ def setUp(self):
...
-        self.gnmicli_hotplug_params = get_gnmi_params_hotplug(self.config_data)
-        print(self.gnmicli_hotplug_params)
+        self.gnmictl_params = get_gnmi_params_simple(self.config_data)
+        self.gnmictl_hotplug_params = get_gnmi_params_hotplug(self.config_data)
+        log.info(self.gnmictl_hotplug_params)
         self.interface_ip_list = get_interface_ipv4_dict(self.config_data)
 
     def runTest(self):
-        if not test_utils.gen_dep_files_p4c_ovs_pipeline_builder(self.config_data):
+        # Compile p4 file using p4c compiler and generate binary using ovs pipeline builder
```
Comment (dkaranjai) [inline tests/DPDK_Hot_Plug_multi_port_test.py]: will this be ovs or tdi ?

File: tests/dpdk_l3em_indirect_counter.py [hunk 2]
```diff
@@ -29,170 +29,229 @@
...
+        self.gnmictl_params = get_gnmi_params_simple(self.config_data)
 
     def runTest(self):
-        
-        if not gnmi_cli_set_and_verify(self.gnmicli_params):
+
+        # Create Ports using gnmictl 
+        if not gnmi_ctl_set_and_verify(self.gnmictl_params):
             self.result.addFailure(self, sys.exc_info())
             self.fail("Failed to configure gnmi cli ports")
```
Comment (dkaranjai) [inline tests/dpdk_l3em_indirect_counter.py:59]: ctl

File: tests/l2_exact_match_with_tap_port.py [hunk 3]
```diff
@@ -60,84 +61,111 @@ def setUp(self):
...
+        if not test_utils.gen_dep_files_p4c_tdi_pipeline_builder(self.config_data):
             self.result.addFailure(self, sys.exc_info())
             self.fail("Failed to generate P4C artifacts or pb.bin")
-
-        if not gnmi_cli_set_and_verify(self.gnmicli_params):
+        
+        # Create Ports using gnmi-ctl
+        if not gnmi_ctl_set_and_verify(self.gnmicli_params):
             self.result.addFailure(self, sys.exc_info())
             self.fail("Failed to configure gnmi cli ports")
```
Comment (dkaranjai) [inline tests/l2_exact_match_with_tap_port.py]: ctl

File: tests/l3_exact_match_with_vhost_add_del.py [hunk 4]
```diff
@@ -64,17 +58,17 @@ def setUp(self):
...
         
-        if not test_utils.gen_dep_files_p4c_ovs_pipeline_builder(self.config_data):
+        if not test_utils.gen_dep_files_p4c_tdi_pipeline_builder(self.config_data):
             self.result.addFailure(self, sys.exc_info())
             self.fail("Failed to generate P4C artifacts or pb.bin")
 
-        if not gnmi_cli_set_and_verify(self.gnmicli_params):
+        if not gnmi_ctl_set_and_verify(self.gnmicli_params):
             self.result.addFailure(self, sys.exc_info())
             self.fail("Failed to configure gnmi cli ports")
```
Comment (dkaranjai) [inline tests/l3_exact_match_with_vhost_add_del.py]: ctl

File: tests/lnt_4vm_on_same_host.py [hunk 5]
```diff
@@ -32,141 +32,173 @@
...
-        if not test_utils.gen_dep_files_p4c_dpdk_pna_ovs_pipeline_builder(self.config_data):
+        if not test_utils.gen_dep_files_p4c_dpdk_pna_tdi_pipeline_builder(
+            self.config_data
+        ):
             self.result.addFailure(self, sys.exc_info())
             self.fail("Failed to generate P4C artifacts or pb.bin")
-        if not gnmi_cli_set_and_verify(self.gnmicli_params):
+        if not gnmi_ctl_set_and_verify(self.gnmicli_params):
             self.result.addFailure(self, sys.exc_info())
             self.fail("Failed to configure gnmi cli ports")
```
Comment (dkaranjai) [inline tests/lnt_4vm_on_same_host.py]: ctl

File: tests/l2_dpdk_port_flapping_with_tap_ports.py [hunk 6]
```diff
@@ -41,6 +42,7 @@ class DpdkPortFlapping(BaseTest):
     to write new testcase, we need to have parent class as BaseTest
     which in turn uses unittest.TestCase, used to create new test cases
     """
+
```
Comment (dkaranjai) [inline tests/l2_dpdk_port_flapping_with_tap_ports.py]: reference to gnmi-cli in multple places

File: tests/l2_dpdk_port_flapping_with_tap_ports.py [hunk 7]
```diff
@@ -309,12 +346,11 @@ def tearDown(self):
         :return: None
```
Comment (dkaranjai) [inline tests/l2_dpdk_port_flapping_with_tap_ports.py]: p4ovs pipeline?

File: tests/l2_exact_match_with_tap_port.py [hunk 8]
```diff
@@ -60,84 +61,111 @@ def setUp(self):
         self.gnmicli_params = get_gnmi_params_simple(self.config_data)
```
Comment (dkaranjai) [inline tests/l2_exact_match_with_tap_port.py]: gnmictl

File: tests/l2_exact_match_with_tap_port.py [hunk 9]
```diff
@@ -60,84 +61,111 @@ def setUp(self):
...
-        if not test_utils.gen_dep_files_p4c_ovs_pipeline_builder(self.config_data):
+         # Compile p4 file using p4c compiler and generate binary using ovs pipeline builder 
+        if not test_utils.gen_dep_files_p4c_tdi_pipeline_builder(self.config_data):
             self.result.addFailure(self, sys.exc_info())
             self.fail("Failed to generate P4C artifacts or pb.bin")
-
-        if not gnmi_cli_set_and_verify(self.gnmicli_params):
+        
+        # Create Ports using gnmi-ctl
+        if not gnmi_ctl_set_and_verify(self.gnmicli_params):
```
Comment (dkaranjai) [inline tests/l2_exact_match_with_tap_port.py]: gnmictl

File: tests/l2_exact_match_with_tap_port.py (hunk 3 above)
Comment (dkaranjai) [inline tests/l2_exact_match_with_tap_port.py]: gnmictl

File: tests/l3_action_profile_with_link_ports.py [hunk 10]
```diff
@@ -59,18 +59,19 @@ def setUp(self):
...
         self.capture_port = test_params['pci_bdf'][:-1] + "1"
         self.config_data = get_config_dict(config_json, test_params['pci_bdf'])
-        self.gnmicli_params = get_gnmi_params_simple(self.config_data)
+        self.gnmictl_params = get_gnmi_params_simple(self.config_data)
         self.interface_ip_list = get_interface_ipv4_dict(self.config_data)
 
 
     def runTest(self):
-        if not test_utils.gen_dep_files_p4c_ovs_pipeline_builder(self.config_data):
+        # Compile p4 file using p4c compiler and generate binary using ovs pipeline builder
```
Comment (dkaranjai) [inline tests/l3_action_profile_with_link_ports.py]: ovs or tdi

File: tests/l3_action_profile_with_tap_ports.py [hunk 11]
```diff
@@ -30,262 +31,347 @@
...
+        # Compile p4 file using p4c compiler and generate binary using ovs pipeline builder
+        if not test_utils.gen_dep_files_p4c_tdi_pipeline_builder(self.config_data):
             self.result.addFailure(self, sys.exc_info())
             self.fail("Failed to generate P4C artifacts or pb.bin")
 
-        if not gnmi_cli_set_and_verify(self.gnmicli_params):
+        # Create Ports using gnmi-ctl
+        if not gnmi_ctl_set_and_verify(self.gnmictl_params):
             self.result.addFailure(self, sys.exc_info())
             self.fail("Failed to configure gnmi cli ports")
```
Comment (dkaranjai) [inline tests/l3_action_profile_with_tap_ports.py:64]: ctl

File: tests/l3_action_selector_with_tap_ports.py [hunk 12]
```diff
@@ -36,435 +37,657 @@
...
+        # Compile p4 file using p4c compiler and generate binary using ovs pipeline builder
+        if not test_utils.gen_dep_files_p4c_tdi_pipeline_builder(self.config_data):
             self.result.addFailure(self, sys.exc_info())
             self.fail("Failed to generate P4C artifacts or pb.bin")
 
-        if not gnmi_cli_set_and_verify(self.gnmicli_params):
+        # Create ports using gnmi-ctl
+        if not gnmi_ctl_set_and_verify(self.gnmictl_params):
             self.result.addFailure(self, sys.exc_info())
             self.fail("Failed to configure gnmi cli ports")
```
Comment (dkaranjai) [inline tests/l3_action_selector_with_tap_ports.py:70]: ctl

File: tests/l3_dpdk_port_flapping_with_vhost_ports.py [hunk 13]
```diff
@@ -171,12 +177,13 @@ def tearDown(self):
         :rtype: None
```
Comment (dkaranjai) [inline tests/l3_dpdk_port_flapping_with_vhost_ports.py]: p4ovs pipeline?

File: tests/l3_exact_match_with_tap_port.py [hunk 14]
```diff
@@ -31,234 +31,318 @@
...
         self.config_data = get_config_dict(config_json)
 
-        self.gnmicli_params = get_gnmi_params_simple(self.config_data)
+        self.gnmictl_params = get_gnmi_params_simple(self.config_data)
         self.interface_ip_list = get_interface_ipv4_dict(self.config_data)
 
-
     def runTest(self):
-        if not test_utils.gen_dep_files_p4c_ovs_pipeline_builder(self.config_data):
+        # Compile p4 file using p4c compiler and generate binary using ovs pipeline builder
```
Comment (dkaranjai) [inline tests/l3_exact_match_with_tap_port.py]: ovs pipeline builder? or tdi?

File: tests/l3_exact_match_with_vhost_add_del.py [hunk 15]
```diff
@@ -64,17 +58,17 @@ def setUp(self):
     def runTest(self):
 
         
-        if not test_utils.gen_dep_files_p4c_ovs_pipeline_builder(self.config_data):
+        if not test_utils.gen_dep_files_p4c_tdi_pipeline_builder(self.config_data):
             self.result.addFailure(self, sys.exc_info())
             self.fail("Failed to generate P4C artifacts or pb.bin")
 
-        if not gnmi_cli_set_and_verify(self.gnmicli_params):
+        if not gnmi_ctl_set_and_verify(self.gnmicli_params):
```
Comment (dkaranjai) [inline tests/l3_exact_match_with_vhost_add_del.py]: ctl

File: tests/l3_exact_match_with_vhost_add_del.py [hunk 16]
```diff
@@ -19,30 +19,24 @@
...
 import common.utils.test_utils as test_utils
-from common.utils.config_file_utils import get_config_dict, get_gnmi_params_simple, get_interface_ipv4_dict
-from common.utils.gnmi_cli_utils import gnmi_cli_set_and_verify, gnmi_set_params, ip_set_ipv4
+from common.utils.config_file_utils import get_config_dict, get_gnmi_params_simple
+from common.utils.gnmi_ctl_utils import gnmi_ctl_set_and_verify
 from common.lib.telnet_connection import connectionManager
+import common.utils.log as log
 
 
 class L3_Exact_Match(BaseTest):
```
Comment (dkaranjai) [inline tests/l3_exact_match_with_vhost_add_del.py]: self.gnmicli_params = get_gnmi_params_simple(self.config_data)
need to change this

File: tests/l3_wcm_with_tap_ports.py [hunk 17]
```diff
@@ -37,99 +37,139 @@
...
 
         self.config_data = get_config_dict(config_json)
 
-        self.gnmicli_params = get_gnmi_params_simple(self.config_data)
+        self.gnmictl_params = get_gnmi_params_simple(self.config_data)
         self.interface_ip_list = get_interface_ipv4_dict(self.config_data)
 
     def runTest(self):
-        if not test_utils.gen_dep_files_p4c_ovs_pipeline_builder(self.config_data):
+        # Compile p4 file using p4c compiler and generate binary using ovs pipeline builder
```
Comment (dkaranjai) [inline tests/l3_wcm_with_tap_ports.py]: ovs pipeline builder?

File: tests/lnt_1hotplugvm_vxlan_1nsvm.py [hunk 18]
```diff
@@ -31,315 +31,416 @@
...
+        vm1result1 = list(dropwhile(lambda x: "lo\r" not in x, result))
+
+        if not gnmi_ctl_utils.gnmi_ctl_set_and_verify(self.gnmictl_params):
             self.result.addFailure(self, sys.exc_info())
             self.fail("Failed to configure gnmi cli ports")
 
-        if not gnmi_cli_utils.gnmi_cli_set_and_verify(self.gnmicli_hotplug_params):
+        if not gnmi_ctl_utils.gnmi_ctl_set_and_verify(self.gnmictl_hotplug_params):
             self.result.addFailure(self, sys.exc_info())
             self.fail("Failed to configure hotplug through gnmi")
```
Comment (dkaranjai) [inline tests/lnt_1hotplugvm_vxlan_1nsvm.py]: ctl

File: tests/lnt_2hotplugvm_vxlan_2nsvm.py [hunk 19]
```diff
@@ -32,330 +32,439 @@
...
         vm2_command_list = ["ip a | egrep \"[0-9]*: \" | cut -d ':' -f 2"]
         result = test_utils.sendCmd_and_recvResult(conn2, vm2_command_list)[0]
         result = result.split("\n")
-        vm2result1 = list(dropwhile(lambda x: 'lo\r' not in x, result))
+        vm2result1 = list(dropwhile(lambda x: "lo\r" not in x, result))
 
-        if not gnmi_cli_utils.gnmi_cli_set_and_verify(self.gnmicli_params):
+        if not gnmi_ctl_utils.gnmi_ctl_set_and_verify(self.gnmictl_params):
             self.result.addFailure(self, sys.exc_info())
             self.fail("Failed to configure gnmi cli ports")
```
Comment (dkaranjai) [inline tests/lnt_2hotplugvm_vxlan_2nsvm.py]: ctl

File: tests/lnt_2hotplugvm_vxlan_2nsvm.py [hunk 20]
```diff
@@ -32,330 +32,439 @@
...
+        self.tap_port_list = gnmi_ctl_utils.get_tap_port_list(self.config_data)
+        self.link_port_list = gnmi_ctl_utils.get_link_port_list(self.config_data)
+        log.info(self.gnmictl_hotplug_params)
         self.interface_ip_list = get_interface_ipv4_dict(self.config_data)
         self.conn_obj_list = []
-        
+
     def runTest(self):
         # ------------------------------------------------- #
         # Configure P4OVS, VM, VLAN adn VXLAN on Local Host #
```
Comment (dkaranjai) [inline tests/lnt_2hotplugvm_vxlan_2nsvm.py]: P4OVS?

File: tests/lnt_2vm_vxlan_2nsvm_5min_ping.py [hunk 21]
```diff
@@ -33,285 +33,382 @@
...
+
+        self.config_data = get_config_dict(
+            config_json,
+            pci_bdf=test_params["pci_bdf"],
+            vm_location_list=test_params["vm_location_list"],
+            vm_cred=self.vm_cred,
+            client_cred=test_params["client_cred"],
+            remote_port=test_params["remote_port"],
+        )
         self.gnmicli_params = get_gnmi_params_simple(self.config_data)
```
Comment (dkaranjai) [inline tests/lnt_2vm_vxlan_2nsvm_5min_ping.py]: gnmictl

File: tests/lnt_2vm_vxlan_2nsvm_5min_ping.py [hunk 22]
```diff
@@ -33,285 +33,382 @@
...
         self.gnmicli_hotplug_params = get_gnmi_params_hotplug(self.config_data)
```
Comment (dkaranjai) [inline tests/lnt_2vm_vxlan_2nsvm_5min_ping.py]: gnmictl

File: tests/lnt_4vm_2host_add_dele_rule.py [hunk 23]
```diff
@@ -31,364 +31,483 @@
...
+
+        self.config_data = get_config_dict(
+            config_json,
+            pci_bdf=test_params["pci_bdf"],
+            vm_location_list=test_params["vm_location_list"],
+            vm_cred=self.vm_cred,
+            client_cred=test_params["client_cred"],
+            remote_port=test_params["remote_port"],
+        )
         self.gnmicli_params = get_gnmi_params_simple(self.config_data)
```
Comment (dkaranjai) [inline tests/lnt_4vm_2host_add_dele_rule.py]: ctl

File: tests/lnt_4vm_2host_add_dele_rule.py [hunk 24]
```diff
@@ -31,364 +31,483 @@
...
-        print ("Begin to configure P4OVS and VM on local host")
-        if not test_utils.gen_dep_files_p4c_dpdk_pna_ovs_pipeline_builder(self.config_data):
+        log.info("Begin to configure P4OVS and VM on local host")
+        if not test_utils.gen_dep_files_p4c_dpdk_pna_tdi_pipeline_builder(
+            self.config_data
+        ):
             self.result.addFailure(self, sys.exc_info())
             self.fail("Failed to generate P4C artifacts or pb.bin")
-        if not gnmi_cli_utils.gnmi_cli_set_and_verify(self.gnmicli_params):
+        if not gnmi_ctl_utils.gnmi_ctl_set_and_verify(self.gnmicli_params):
```
Comment (dkaranjai) [inline tests/lnt_4vm_2host_add_dele_rule.py]: gnmicli_params -> gnmictl

File: tests/lnt_4vm_on_same_host.py [hunk 25]
```diff
@@ -32,141 +32,173 @@
...
         except KeyError:
             self.vm_cred = ""
 
-        self.config_data = get_config_dict(config_json,vm_location_list=test_params['vm_location_list'],vm_cred=self.vm_cred)
+        self.config_data = get_config_dict(
+            config_json,
+            vm_location_list=test_params["vm_location_list"],
+            vm_cred=self.vm_cred,
+        )
         self.gnmicli_params = get_gnmi_params_simple(self.config_data)
```
Comment (dkaranjai) [inline tests/lnt_4vm_on_same_host.py]: gnmictl

File: tests/lnt_4vm_on_same_host.py [hunk 26]
```diff
@@ -32,141 +32,173 @@
...
+
     def runTest(self):
-        if not test_utils.gen_dep_files_p4c_dpdk_pna_ovs_pipeline_builder(self.config_data):
+        if not test_utils.gen_dep_files_p4c_dpdk_pna_tdi_pipeline_builder(
+            self.config_data
+        ):
             self.result.addFailure(self, sys.exc_info())
             self.fail("Failed to generate P4C artifacts or pb.bin")
-        if not gnmi_cli_set_and_verify(self.gnmicli_params):
+        if not gnmi_ctl_set_and_verify(self.gnmicli_params):
```
Comment (dkaranjai) [inline tests/lnt_4vm_on_same_host.py]: gnmictl

**Review by dkaranjai — CHANGES_REQUESTED** [review]
Also please check the comments once. noticed a few discrepencies:
1. no space after "#"
2. spelling mistakes like "2th"

**Review by stadju — APPROVED** [review]

File: tests/l2_exact_match_with_tap_port.py (hunk 3 above)
Comment (sjwahub) [reply tests/l2_exact_match_with_tap_port.py]: resolved

File: tests/dpdk_l3em_indirect_counter.py (hunk 2 above)
Comment (sjwahub) [reply tests/dpdk_l3em_indirect_counter.py:59]: resolved

File: tests/lnt_2vm_vxlan_2nsvm_5min_ping.py (hunk 22 above)
Comment (sjwahub) [reply tests/lnt_2vm_vxlan_2nsvm_5min_ping.py]: fixed

File: tests/lnt_4vm_2host_add_dele_rule.py (hunk 23 above)
Comment (sjwahub) [reply tests/lnt_4vm_2host_add_dele_rule.py]: fixed

File: tests/DPDK_Hot_Plug_multi_port_test.py (hunk 1 above)
Comment (sjwahub) [reply tests/DPDK_Hot_Plug_multi_port_test.py]: This is a merge issue. At my end, it looks good

File: tests/l3_exact_match_with_vhost_add_del.py (hunk 4 above)
Comment (sjwahub) [reply tests/l3_exact_match_with_vhost_add_del.py]: fixed

File: tests/lnt_4vm_2host_add_dele_rule.py (hunk 24 above)
Comment (sjwahub) [reply tests/lnt_4vm_2host_add_dele_rule.py]: fixed

File: tests/lnt_4vm_on_same_host.py (hunk 25 above)
Comment (sjwahub) [reply tests/lnt_4vm_on_same_host.py]: fixed

File: tests/lnt_4vm_on_same_host.py (hunk 26 above)
Comment (sjwahub) [reply tests/lnt_4vm_on_same_host.py]: fixed

File: tests/l2_dpdk_port_flapping_with_tap_ports.py (hunk 6 above)
Comment (sjwahub) [reply tests/l2_dpdk_port_flapping_with_tap_ports.py]: fixed

File: tests/l2_dpdk_port_flapping_with_tap_ports.py (hunk 7 above)
Comment (sjwahub) [reply tests/l2_dpdk_port_flapping_with_tap_ports.py]: fixed

File: tests/l2_exact_match_with_tap_port.py (hunk 8 above)
Comment (sjwahub) [reply tests/l2_exact_match_with_tap_port.py]: fixed

File: tests/l2_exact_match_with_tap_port.py (hunk 3 above)
Comment (sjwahub) [reply tests/l2_exact_match_with_tap_port.py]: fixed

File: tests/l3_action_profile_with_link_ports.py (hunk 10 above)
Comment (sjwahub) [reply tests/l3_action_profile_with_link_ports.py]: merge issue ? No issue at my end

File: tests/l3_action_profile_with_tap_ports.py (hunk 11 above)
Comment (sjwahub) [reply tests/l3_action_profile_with_tap_ports.py:64]: fixed

File: tests/l2_exact_match_with_tap_port.py (hunk 9 above)
Comment (sjwahub) [reply tests/l2_exact_match_with_tap_port.py]: fixed

File: tests/l3_action_selector_with_tap_ports.py (hunk 12 above)
Comment (sjwahub) [reply tests/l3_action_selector_with_tap_ports.py:70]: fixed

File: tests/l3_dpdk_port_flapping_with_vhost_ports.py (hunk 13 above)
Comment (sjwahub) [reply tests/l3_dpdk_port_flapping_with_vhost_ports.py]: fixed

File: tests/l3_exact_match_with_tap_port.py (hunk 14 above)
Comment (sjwahub) [reply tests/l3_exact_match_with_tap_port.py]: fixed

File: tests/l3_exact_match_with_vhost_add_del.py (hunk 15 above)
Comment (sjwahub) [reply tests/l3_exact_match_with_vhost_add_del.py]: fixed

File: tests/l3_exact_match_with_vhost_add_del.py (hunk 16 above)
Comment (sjwahub) [reply tests/l3_exact_match_with_vhost_add_del.py]: changed

File: tests/l3_wcm_with_tap_ports.py (hunk 17 above)
Comment (sjwahub) [reply tests/l3_wcm_with_tap_ports.py]: fixed

File: tests/lnt_1hotplugvm_vxlan_1nsvm.py (hunk 18 above)
Comment (sjwahub) [reply tests/lnt_1hotplugvm_vxlan_1nsvm.py]: fixed

File: tests/lnt_2hotplugvm_vxlan_2nsvm.py (hunk 19 above)
Comment (sjwahub) [reply tests/lnt_2hotplugvm_vxlan_2nsvm.py]: fxied

File: tests/lnt_2hotplugvm_vxlan_2nsvm.py (hunk 20 above)
Comment (sjwahub) [reply tests/lnt_2hotplugvm_vxlan_2nsvm.py]: fixed

File: tests/DPDK_Hot_Plug_multi_port_test.py (hunk 1 above)
Comment (sjwahub) [reply tests/DPDK_Hot_Plug_multi_port_test.py]: clone this branch and fixed it

File: tests/lnt_4vm_on_same_host.py (hunk 5 above)
Comment (sjwahub) [reply tests/lnt_4vm_on_same_host.py]: fixed

**Review by dkaranjai — APPROVED** [review]

**Comment by dkaranjai** [comment]
Changes moved to PR #3

**Pull request closed without merging**